src/web-worker/amp-worker.js

```diff
@@ -69,11 +70,25 @@ class AmpWorker {
     /** @const @private {!Window} */
     this.win_ = win;
 
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(win);
+
     const url =
         calculateEntryPointScriptUrl(location, 'ww', getMode().localDev);
-    /** @const @private {!Worker} */
-    this.worker_ = new win.Worker(url);
-    this.worker_.onmessage = this.receiveMessage_.bind(this);
+    dev().fine(TAG, 'Fetching web worker from:', url);
+
+    /** @private {Worker} */
+    this.worker_ = null;
+
+    /** @const @private {!Promise} */
+    this.fetchPromise_ = this.xhr_.fetchText(url).then(text => {
+      // Fetch web worker binary and create from Blob as workaround since
+      // Worker constructor won't accept CORS URLs.
+      const blob = new win.Blob([text], {type: 'text/javascript'});
+      const blobUrl = win.URL.createObjectURL(blob);
+      this.worker_ = new win.Worker(blobUrl);
```

Comment:
Good call -- actually it does violate our CSP after this PR due to Blob-ification. 

`worker-src` is CSP3 which has [poor support](https://www.chromestatus.com/features/5922594955984896). Going to add `blob:` to fallback `default-src` instead: cl/147751445